frontend/src/components/widgets/CustomComponent/ComponentInstance.test.tsx

```diff
@@ -173,6 +180,66 @@ describe("ComponentInstance", () => {
     expect(iframe.prop("sandbox")).toEqual(DEFAULT_IFRAME_SANDBOX_POLICY)
   })
 
+  it("should send JSON args to iframe", () => {
+    const mc = new MockComponent()
+
+    // We should receive an initial RENDER message with no arguments
+    mc.sendBackMsg(ComponentMessageType.COMPONENT_READY, { apiVersion: 1 })
+    expect(mc.receiveForwardMsg).toHaveBeenLastCalledWith(
+      renderMsg({}, []),
+      "*"
+    )
+
+    // Simulate passing JSON arguments to the component.
+    const jsonArgs = { foo: "string", bar: 5 }
+    const element = createElementProp(jsonArgs, [])
+    mc.wrapper.setProps({ element })
+
+    expect(mc.instance.state.componentError).toBeUndefined()
+
+    // We should get those JSON arguments in our receiveForwardMsg callback.
+    expect(mc.receiveForwardMsg).toHaveBeenLastCalledWith(
+      renderMsg(jsonArgs, []),
+      "*"
+    )
+  })
+
+  it("should send bytes args to iframe", () => {
+    const mc = new MockComponent()
+
+    // We should receive an initial RENDER message with no arguments
+    mc.sendBackMsg(ComponentMessageType.COMPONENT_READY, { apiVersion: 1 })
+    expect(mc.receiveForwardMsg).toHaveBeenLastCalledWith(
+      renderMsg({}, []),
+      "*"
+    )
+
+    // Bytes are passed from the backend as "SpecialArgs".
+    const key1 = "bytes1"
+    const bytes1 = new Uint8Array([0, 1, 2, 3])
+    const key2 = "bytes2"
+    const bytes2 = new Uint8Array([4, 5, 6, 7])
+    const element = createElementProp({}, [
+      SpecialArg.create({ key: key1, bytes: bytes1 }),
+      SpecialArg.create({ key: key2, bytes: bytes2 }),
+    ])
+    mc.wrapper.setProps({ element })
+
+    expect(mc.instance.state.componentError).toBeUndefined()
+
+    // The iframe receives bytes in its args dict, alongside JSON.
+    // The funny `[key1]: bytes1` syntax is because these are computed
+    // property names (and not string literals).
+    expect(mc.receiveForwardMsg).toHaveBeenLastCalledWith(
+      renderMsg({ [key1]: bytes1, [key2]: bytes2 }, []),
+      "*"
+    )
+  })
+
+  it("should send dataframe args to iframe", () => {
+    // TODO for Henrikh
```

Comment:
FIne with this? Is there an issue/task somewhere so that he's aware of what to do?

Reply:
Yep yep, he's aware - there's another TODO for him in this file as well (cc @kantuni)